modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -663,37 +614,24 @@ module ChapelIteratorSupport {
   }
 
   proc chpl__dynamicFastFollowCheck(x, lead) {
-    return chpl__hasInertFastFollowers(x);
-  }
-
-  proc chpl__dynamicFastFollowCheck(x: [], lead) {
-    if chpl__staticFastFollowCheck(x, lead) then
-      return x._value.dsiDynamicFastFollowCheck(lead);
-    else
-      return false;
-  }
-
-  proc chpl__dynamicFastFollowCheckZip(x: _tuple) {
-    if !chpl__canHaveFastFollowersZip(x) {
-      return false;
-    }
-
-    if chpl__canLeadFastFollowers(x(0)) {
-      return chpl__dynamicFastFollowCheckZip(x, x(0));
+    if chpl__canLeadFastFollowers(lead) {
+      if (isArray(x)) {
+        if chpl__staticFastFollowCheck(x, lead) {
+          return x._value.dsiDynamicFastFollowCheck(lead);
+        }
+        else {
+          return false;
+        }
+      }
+      else {
+        return chpl__hasInertFastFollowers(x);
```

Comment:
If `chpl__hasInertFastFollowers` is now simply !isArray, then both its remaining uses can be converted to `return true` and chpl__hasInertFastFollowers can be eliminated altogether. I do not see an intrinsic value in having this function.